Loop/Managers/Alerts/AlertManager.swift

```diff
@@ -260,6 +272,43 @@ public final class AlertManager {
                      trigger: .delayed(interval: .hours(24)))
     }
 
+    // MARK: - Rescheduling Muted Alerts
+
+    func rescheduleMutedAlerts(_ newValue: AlertMuter.Configuration) {
+        UserDefaults.standard.alertMuterConfiguration = newValue
+
+        rescheduleLoopNotRunningNotifications()
+
+        let shouldMuteAlerts = newValue.shouldMuteAlerts
+        lookupAllPendingDelayedOrRepeatingAlerts() { [weak self] result in
+            switch result {
+            case .success(let persistedAlerts):
+                for persistedAlert in persistedAlerts {
+                    if shouldMuteAlerts {
+                        // reschedule a muted alert, if needed
+                        switch persistedAlert.alert.trigger {
+                        case .delayed(let interval), .repeating(let interval):
+                            let timeFromNow = persistedAlert.issuedDate.advanced(by: interval).timeIntervalSinceNow
+                            if newValue.shouldMuteAlert(timeFromNow) {
+                                self?.retractAlert(identifier: persistedAlert.alert.identifier)
```

Comment:
This seems to me to be using the wrong level of abstraction.  The alerts aren't retracted and then reissued. Any future-scheduled *issued* alerts remain issued, but are rescheduled with silence, like is being done above with the loop-not-looping alerts. This only impacts the system notifications, as they're the only ones that are "scheduled". Local modal presentation can check the AlertMuter state on presentation.